src/main/test/com/epam/preprod/sirenko/strategy/impl/AutoGeneratedProductCreationStrategyTest.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.preprod.sirenko.strategy.impl;
+
+import com.epam.preprod.sirenko.enums.PetGroup;
+import com.epam.preprod.sirenko.enums.Season;
+import com.epam.preprod.sirenko.enums.Size;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AutoGeneratedProductCreationStrategyTest {
+	private AutoGeneratedProductCreationStrategy autoGeneratedProduct;
+	
+	@BeforeEach
+	public void SetData() {
+		autoGeneratedProduct = mock(AutoGeneratedProductCreationStrategy.class);
```

Comment:
Its wrong approach to test your class. You should not mock test object. You should mock external components of your class (like some other class that is a field in your test object)